web/webkit/src/main/scala/net/liftweb/http/LiftSession.scala

```diff
@@ -2568,12 +2569,14 @@ class LiftSession(private[http] val _contextPath: String, val underlyingId: Stri
   }
 
   // Given a comet creation info, build a comet based on the comet type, first
-  // attempting to use LiftRules.cometCreationFactory and then building it by
+  // attempting to use `LiftRules.cometCreationFactory` and then attempting to
+  // find a match in `LiftRules.cometCreation`. Failing those, this will build it by
   // class name. Return a descriptive Failure if it's all gone sideways.
   //
   // Runs some base setup tasks before returning the comet.
   private def buildCometByCreationInfo(creationInfo: CometCreationInfo): Box[LiftCometActor] = {
-    LiftRules.cometCreationFactory.vend.apply(creationInfo) or {
+    LiftRules.cometCreationFactory.vend.apply(creationInfo) or
+    LiftRules.cometCreation.toList.find(_.isDefinedAt(creationInfo)).map(_.apply(creationInfo)) or {
```

Comment:
Wonder if we could use `NamedPF.applyBox(creationInfo, LiftRules.cometCreation.toList)` here.